### PR TITLE
k8s_events: add events module

### PR DESCRIPTION
**Pull request by EmilyM1**
##### SUMMARY
Adds a module for creation on Kubernetes Events

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
k8s_events

### CODE REVIEW

**Comment by geerlingguy** [comment]
@EmilyM1 - The Kubernetes-related modules and plugins have been moved to a new `community.kubernetes` collection, where further development and module additions (like this one) will go. Would you be willing/able to create a new PR adding this module to that project?

See related issue: https://github.com/ansible-collections/kubernetes/issues/8

File: lib/ansible/modules/clustering/k8s/k8s_event.py [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+        prior_count = 1
+        now = datetime.datetime.now()
+        rfc = kubernetes.config.dateutil.format_rfc3339(now)
+        first_timestamp = rfc
+        last_timestamp = rfc
+
+        if prior_event and prior_event["reason"] == reason:
+            prior_count = prior_event["count"] + 1
+            first_timestamp = prior_event["firstTimestamp"]
+            last_timestamp = rfc
```
Comment (manojranak1) [inline lib/ansible/modules/clustering/k8s/k8s_event.py:296]: redundant code to line 291

File: lib/ansible/modules/clustering/k8s/k8s_event.py [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+
+        meta = definition["metadata"]
+        meta["name"] = self.params.get("name")
+        meta["namespace"] = self.params.get("namespace")
+
+        involved_obj = self.params.get("involvedObject")
+
+        resource = self.find_resource("Event", "v1", fail=True)
+
+        prior_event = None
```
Comment (manojranak1) [inline lib/ansible/modules/clustering/k8s/k8s_event.py:278]: This line of code is not required, you can modify your condition @ line 293 as "prior_event.get('reason') == reason"

**Comment by Akasurde** [comment]
@EmilyM1 Are you still working on this? If yes, could you please migrate this module to [`community.kubernetes`](https://github.com/ansible-collections/kubernetes) collection. Since every plugin and modules related to Kubernetes is migrated there.

Thanks.

needs_info

**Pull request closed without merging**